docs/INSTALL.md

```diff
@@ -0,0 +1,659 @@
+
+<a name="recommended-setup-for-running-ilias"></a>
+## Recommended Setup for Running ILIAS
+
+For best results we RECOMMEND:
+
+  * Debian GNU Linux 8 / Red Hat Enterprise Linux 7 / Ubuntu 16.04 LTS
+  * MySQL 5.5 / MariaDB
+  * PHP 5.6+
+  * Apache 2.2+ with mod_php
+  * ImageMagick 6.x+
+  * php5-gd, php5-xsl, php5-mysql
+  * OpenJDK 7+
+  * zip, unzip
+  * git
+
+<a name="supported-platforms"></a>
+## Supported Platforms
+
+Please note that different configurations SHOULD be possible, but it might be harder to find someone who can help when things go south. You SHALL NOT use a different configuration unless you are an experienced system administrator.
+
+<a name="server"></a>
+### Server
+
+  * Server OS: Linux
+  * Web Server: Apache 2 (mod_php, php-fpm)
+  * Databases: MySQL/MariaDB 5.0+ and Galera (experimental), Oracle 10g+ (experimental), PostgreSQL (experimental)
+  * PHP: Version 5.5+ and 7.0+ are supported
+  
+See http://www.ilias.de/docu/goto.php?target=lm_367&client_id=docu for details.
+  
+<a name="client"></a>
+### Client
+
+  * Desktop: Windows 7+, MacOS X 10.7+, Linux
+  * Web Browser: IE11+, Microsoft Edge, Firefox 14+, Chrome 18+, Safari 7+
+
+<a name="database-recommendations"></a>
+## Database Recommendations
+
+We RECOMMEND to use MySQL/MariaDB with the following settings:
+
+  * InnoDB storage engine
+  * utf8_general_ci
+  * query_cache_size (> 16M)
+  * join_buffer_size (> 128.0K, or always use indexes with joins)
+  * table_open_cache (> 400)
+  * innodb_buffer_pool_size (>= 2G, depending on DB size)
+
+On MySQL 5.6 the setting ```STRICT_TRANS_TABLES``` or ```STRICT_ALL_TABLES``` MUST be disabled.
+
+<a name="manual-installation-on-linux"></a>
+# Manual Installation on Linux
+
+You can download the latest ILIAS release at http://www.ilias.de/docu/goto.php?target=st_229 or clone it from GitHub at https://github.com/ILIAS-eLearning/ILIAS (for production make sure to checkout the latest stable release, not trunk).
+
+We RECOMMEND to clone from GitHub as this will offer some kind of autoupdate for future releases and versions.
+
+  * Install dependencies (see [Recommended Setup for Running ILIAS](#recommended-setup-for-running-ilias))
+  * Untar/Clone ILIAS into the web servers docroot (e.g. /var/www/html/)
+  * Create directory outside the web servers docroot (e.g. /var/www/files/)
+  * Change owner/group to www-data (on Debian) or apache (on RHEL) for the files and directories created above
+
+<a name="git-clonecheckout"></a>
+## Git Clone/Checkout
+
+To checkout the ILIAS release 5.2 in ```/var/www/html/ilias/``` use the following commands:
+
+```
+cd /var/www/html/
+git clone https://github.com/ILIAS-eLearning/ILIAS.git ilias
+cd ilias
+git checkout release_5-2
+chown www-data:www-data /var/www/html/ilias -R
+```
+The files MUST be owned by your webserver user/group (e.g. ```www-data``` or ```apache```) the mode SHOULD be 644 for files and 755 for directories.
```

Comment:
It is bad practice to set the PHP files to be writable by the web user. This allows an attacker to modify the source files if he gains control over the web server processes. To limit the possible damage, the PHP files should be owned by root wherever possible. The only files and directories that must be owned by the web user are:
* ilias.ini.php
* data/

All the other files and directories should be owned by root, but readable by the web user (e.g. 644/755). The simplest way to achieve that, is to remove the `--recursive` (or `-R`) flag from the `chown` command above. This will allow the setup to create/modify the necessary files and leaves the ownership of the rest as is (root).